### PR TITLE
Add new type of App's extensions' targets

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds more enums for defining places where App's extensions can be mounted. It also introduce changes in the way of handling App extensions. Instead of having 3 enums and handling a different combinations of them, we convert them to single enum.
New `mount` types:
- mount: 
    - NAVIGATION_CATALOG
    - NAVIGATION-ORDERS
    - NAVIGATION_CUSTOMERS
    - NAVIGATION_DISCOUNTS
    - NAVIGATION_PAGES
    - NAVIGATION_TRANSLATIONS
- target:
    - POPUP
    - APP_PAGE

PR with updated docs: https://github.com/saleor/saleor-docs/pull/348

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/app/installation_utils.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def install_app(
             view=extension_data.get("view"),
             type=extension_data.get("type"),
             target=extension_data.get("target"),
+            open_as=extension_data.get("open_as") or AppExtensionOpenAs.POPUP,
```
Comment (fowczarek) [inline saleor/app/installation_utils.py:63]: Maybe?
```suggestion
            open_as=extension_data.get("open_as",  AppExtensionOpenAs.POPUP),
```

File: saleor/graphql/app/types.py [hunk 2]
```diff
@@ -51,6 +54,31 @@ class AppManifestExtension(graphene.ObjectType):
...
+            - appUrl is defined
+        """
+        open_as = root.get("open_as", AppExtensionOpenAs.POPUP)
+        app_url = root["app_url"]
+        url = root["url"]
+        if url.startswith("/") and app_url and open_as == AppExtensionOpenAs.POPUP:
+            parsed_url = urlparse(app_url)
+            new_path = urljoin(parsed_url.path, url[1:])
+            return parsed_url._replace(path=new_path).geturl()
+        return url
```
Comment (fowczarek) [inline saleor/graphql/app/types.py:81]: Maybe we could move it to `resolvers.py` to easier reuse it in `AppExtension.resolve_url`.

**Review by fowczarek — COMMENTED** [review]
Missing changelog.

File: saleor/app/manifest_validations.py [hunk 3]
```diff
@@ -11,34 +12,56 @@
...
+    - url can start with '/' when one of these conditions is true:
+        a) extension.open_as == APP_PAGE
+        b) appUrl is provided
+    - url cannot start with protocol when openAs == "APP_PAGE"
+    """
+    extension_url = extension["url"]
+    open_as = extension.get("open_as") or AppExtensionOpenAs.POPUP
+    if extension_url.startswith("/"):
+        if open_as == AppExtensionOpenAs.APP_PAGE:
+            pass
```
Comment (L3str4nge) [inline saleor/app/manifest_validations.py:47]: I think this statement is redundant. If we want to keep this anyway I would change it to `return`

**Review by L3str4nge — APPROVED** [review]

**Review by jakubkuc — APPROVED** [review]

File: saleor/app/manifest_validations.py (hunk 3 above)
Comment (IKarbowiak) [reply saleor/app/manifest_validations.py:47]: I agree. IMO we should change conditions in some way, or even move all these parts to a separate function.

File: saleor/graphql/app/types.py [hunk 4]
```diff
@@ -51,6 +54,31 @@ class AppManifestExtension(graphene.ObjectType):
     target = AppExtensionTargetEnum(
         description="Place where extension's iframe will be mounted.", required=True
     )
+    open_as = AppExtensionOpenAsEnum(
+        description="Type of way how app extension will be opened.", required=True
+    )
+
+    @staticmethod
+    def resolve_open_as(root, info):
+        return root.get("open_as") or AppExtensionOpenAs.POPUP
```
Comment (IKarbowiak) [inline saleor/graphql/app/types.py:63]:
```suggestion
        return root.get("open_as", AppExtensionOpenAs.POPUP)
```

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/app/types.py (hunk 4 above)
Comment (fowczarek) [reply saleor/graphql/app/types.py:63]: Duplicate:
https://github.com/saleor/saleor/pull/8959#discussion_r785810611

File: saleor/app/installation_utils.py (hunk 1 above)
Comment (korycins) [reply saleor/app/installation_utils.py:63]: In case when extension_data will have key `open_as` with `None` this will break

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

File: saleor/graphql/schema.graphql [hunk 5]
```diff
@@ -317,23 +316,25 @@ type AppExtensionCountableEdge {
...
+  NAVIGATION_DISCOUNTS
+  NAVIGATION_TRANSLATIONS
+  NAVIGATION_PAGES
 }
 
-enum AppExtensionViewEnum {
-  PRODUCT
+enum AppExtensionTargetEnum {
+  POPUP
+  APP_PAGE
```
Comment (maarcingebala) [inline saleor/graphql/schema.graphql:337]: Was it called "Open As" before and now we changed it to target? I think "open as" is a better name for this setting.

Comment (korycins) [reply saleor/graphql/schema.graphql:337]: That was a suggestion added by @jwm0  as he was handling the concept of changes. He said that `target` is popular according to popups

Comment (maarcingebala) [reply saleor/graphql/schema.graphql:337]: OK, if this is approved by frontend then I'm fine with this 👍

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**